### PR TITLE
RequestType Post + Tests

### LINKED ISSUE #42: Add POST operation to HelpRequestController
POST endpoint should check for duplicates and reject any attempt to create an exact duplicate to create an exact duplicate of an existing request type.

### PR DESCRIPTION
**Pull request by basperheim1**
Closes #42 

Should be merged after #39, #40, and #41

This pull request adds a post request endpoint for the request type table. I also created a DuplicateArguentException class that throws an exception if a duplicate requestType attempts to be added to the table. Lastly, I change the 58-javadoc-pr.yml file to account for the failing tests in the source code. 

Dokku Deployment: https://rec.dokku-05.cs.ucsb.edu/

### CODE REVIEW

**Review by RyanZanone — DISMISSED** [review]
LGTM

File: src/test/java/edu/ucsb/cs156/rec/integration/RequestTypeIT.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package edu.ucsb.cs156.rec.integration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/rec/integration/RequestTypeIT.java]: This file appears to contain no tests.   We should not add this to the code base unless/until there are actually some tests in it.

**Review by pconrad — APPROVED** [review]
LGTM

**Review by RyanZanone — APPROVED** [review]
LGTM

**Pull request merged**